src/flint/types/fq_default.pyx

```diff
@@ -0,0 +1,886 @@
+from flint.pyflint cimport global_random_state
+from flint.types.fmpz cimport fmpz, any_as_fmpz
+from flint.types.nmod cimport nmod
+from flint.types.fmpz_mod cimport fmpz_mod
+from flint.types.fmpz_poly cimport fmpz_poly, any_as_fmpz_poly, fmpz_poly_set_list
+from flint.types.fmpz_mod_poly cimport fmpz_mod_poly, fmpz_mod_poly_ctx
+from flint.types.nmod_poly cimport nmod_poly
+from flint.utils.typecheck cimport typecheck
+
+cdef class fq_default_ctx:
+    r"""
+    Context object for creating :class:`~.fq_default`.
+
+    Finite fields can be initialized in one of two possible ways. The
+    first is by providing characteristic and degree:
+
+        >>> fq_default_ctx(5, 2, 'y', fq_type=1)
+        fq_default_ctx(5, 2, 'y', x^2 + 4*x + 2, 'FQ_ZECH')
```

Comment:
Probably better to use `fq_type='FQ_ZECH'` or `fq_type=fq_default_type.FQ_ZECH` here rather than `fq_type=1`. We shouldn't encourage writing code that uses magic values like that.